tests/providers/common/io/xcom/test_backend.py

```diff
@@ -86,28 +87,19 @@ def task_instance(task_instance_factory):
 
 
 class TestXcomObjectStoreBackend:
-    path = "file:/tmp/xcom"
-
-    def setup_method(self):
-        try:
-            conf.add_section("common.io")
-        except DuplicateSectionError:
-            pass
-        conf.set("core", "xcom_backend", "airflow.providers.common.io.xcom.backend.XComObjectStoreBackend")
-        conf.set("common.io", "xcom_objectstore_path", self.path)
-        conf.set("common.io", "xcom_objectstore_threshold", "50")
-        settings.configure_vars()
-
-    def teardown_method(self):
-        conf.remove_option("core", "xcom_backend")
-        conf.remove_option("common.io", "xcom_objectstore_path")
-        conf.remove_option("common.io", "xcom_objectstore_threshold")
-        settings.configure_vars()
-        p = ObjectStoragePath(self.path)
-        if p.exists():
-            p.rmdir(recursive=True)
-
-    @pytest.mark.db_test
+    @pytest.fixture(autouse=True)
+    def setup_test_cases(self, tmp_path):
+        xcom_path = tmp_path / "xcom"
+        xcom_path.mkdir()
+        self.path = f"file:{xcom_path}"
```

Comment:
@bolkedebruin I'm not sure is it a bug or is it by design

When `[common.io] xcom_objectstore_path` set to **file:/foo/bar** XComBackend use absolute path `/foo/bar`
When `[common.io] xcom_objectstore_path` set to **file://foo/bar** XComBackend use relative to the current directory path `foo/bar`
When `[common.io] xcom_objectstore_path` set to **file:///foo/bar** XComBackend failed with an error `ValueError: Invalid key`

Reply:
I *think* that is expected behavior but I need to check.

Reply:
IMHO

> 1) When [common.io] xcom_objectstore_path set to file:/foo/bar XComBackend use absolute path /foo/bar
> 2) When [common.io] xcom_objectstore_path set to file://foo/bar XComBackend use relative to the current directory path foo/bar
> 3) When [common.io] xcom_objectstore_path set to file:///foo/bar XComBackend failed with an error ValueError: Invalid key

1) 2) are ok. 1) is perfectly fine according to RFC3986 and explained nicely here https://en.wikipedia.org/wiki/File_URI_scheme 2) is not correct (`://` should be followed by eiter '/' or literal localhost,) but many implementation accept it as a shortcut

3) invalid key should be accepted - it's fully valid according to RFC and many implementations and should be `/foo/bar` absolute).

But maybe it's been fixed in the last iteration already ?

Reply:
More common URI `file:///foo/bar` seems like now worked fine